internal/stream/stream.go

```diff
@@ -215,16 +214,19 @@ const (
 )
 
 func (w *Writer) flushChunk(last bool) error {
-	if !last && len(w.unwritten) != ChunkSize {
+	if !last && w.unwritten.Len() < ChunkSize {
 		panic("stream: internal error: flush called with partial chunk")
 	}
 
 	if last {
 		setLastChunkFlag(&w.nonce)
 	}
-	buf := w.a.Seal(w.buf[:0], w.nonce[:], w.unwritten, nil)
-	_, err := w.dst.Write(buf)
-	w.unwritten = w.buf[:0]
+
+	chunk := w.unwritten.Next(ChunkSize)
```

Comment:
This doesn't release the memory held by the chunk in the buffer. I think you need a `w.unwritten.Reset()` after this. Otherwise the Writes to `w.unwritten` will continue to grow the buffer, which will cause the entire stream content to stay in memory until the Writer is GC'd.